### PR TITLE
Adding ConfigMap build sources to api

### PR DESCRIPTION
**Pull request by adambkaplan**
* New type `ConfigMapBuildSource`
* Added ConfigMap collections to `CommonSpec`
* Regenerated protobuf, swagger, and deepcopy.

Trello card: https://trello.com/c/RMKJxJUm/1020-5-allow-using-a-configmap-as-an-input-to-a-build-builds

### CODE REVIEW

**Comment by adambkaplan** [comment]
ping @openshift/api-review

File: build/v1/types.go [hunk 1]
```diff
@@ -462,15 +466,32 @@ type ImageSourcePath struct {
 	DestinationDir string `json:"destinationDir" protobuf:"bytes,2,opt,name=destinationDir"`
 }
 
+// LocalObjectBuildSource is a build source that is copied into a build from a Kubernetes
+// key-value store, such as a `Secret` or `ConfigMap`.
+type LocalObjectBuildSource interface {
```
Comment (liggitt) [inline build/v1/types.go]: we typically don't declare interfaces or methods on API types... if you want to adapt them like this, do so in application code, not in the API structs themselves

File: build/v1/types.go [hunk 2]
```diff
@@ -462,15 +466,32 @@ type ImageSourcePath struct {
...
+	// IsSecret returns `true` if the build source is a `Secret` containing sensitive data.
+	IsSecret() bool
+}
+
 // SecretBuildSource describes a secret and its destination directory that will be
 // used only at the build time. The content of the secret referenced here will
 // be copied into the destination directory instead of mounting.
 type SecretBuildSource struct {
-	// secret is a reference to an existing secret that you want to use in your
+	// Secret is a reference to an existing secret that you want to use in your
```
Comment (liggitt) [inline build/v1/types.go]: revert this, comments follow the serialized names

File: build/v1/types.go [hunk 3]
```diff
@@ -462,15 +466,32 @@ type ImageSourcePath struct {
...
 	// build.
 	Secret corev1.LocalObjectReference `json:"secret" protobuf:"bytes,1,opt,name=secret"`
 
-	// destinationDir is the directory where the files from the secret should be
+	// DestinationDir is the directory where the files from the secret should be
```
Comment (liggitt) [inline build/v1/types.go]: revert

File: build/v1/types.go [hunk 4]
```diff
@@ -481,6 +502,67 @@ type SecretBuildSource struct {
...
+// IsSecret returns `true` if the build source is a `Secret` containing sensitive data.
+func (s SecretBuildSource) IsSecret() bool {
+	return true
+}
+
+// ConfigMapBuildSource describes a configmap and its destination directory that will be
+// used only at the build time. The content of the configmap referenced here will
+// be copied into the destination directory instead of mounting.
+type ConfigMapBuildSource struct {
+	// ConfigMap is a reference to an existing configmap that you want to use in your
```
Comment (liggitt) [inline build/v1/types.go]: lowercase comments to match serialized names (used to generate API docs)

File: build/v1/types.go [hunk 5]
```diff
@@ -618,6 +700,9 @@ type CustomBuildStrategy struct {
 
 	// buildAPIVersion is the requested API version for the Build object serialized and passed to the custom builder
 	BuildAPIVersion string `json:"buildAPIVersion,omitempty" protobuf:"bytes,7,opt,name=buildAPIVersion"`
+
+	// Configs is a list of additional configMaps that will be included in the custom build pod
+	Configs []ConfigSpec `json:"configs,omitempty" protobuf:"bytes,8,rep,name=configs"`
```
Comment (liggitt) [inline build/v1/types.go]: `ConfigMaps`?

File: build/v1/types.go [hunk 6]
```diff
@@ -416,6 +416,10 @@ type BuildSource struct {
 	// secrets represents a list of secrets and their destinations that will
 	// be used only for the build.
 	Secrets []SecretBuildSource `json:"secrets,omitempty" protobuf:"bytes,8,rep,name=secrets"`
+
+	// configs represents a list of configmaps and their destinations that will
+	// be used for the build.
+	Configs []ConfigMapBuildSource `json:"configs,omitempty" protobuf:"bytes,9,rep,name=configs"`
```
Comment (liggitt) [inline build/v1/types.go]: `ConfigMaps`?

File: build/v1/types.go (hunk 5 above)
Comment (adambkaplan) [reply build/v1/types.go]: I had that debate myself - I'm OK with that convention.

File: build/v1/types.go (hunk 1 above)
Comment (adambkaplan) [reply build/v1/types.go]: I wanted to keep this interface within Origin proper, but most of the builder code acts upon API structs and not converted internal representations - see https://github.com/openshift/origin/blob/master/pkg/build/builder/docker.go as an example. Exposing this interface at the API level allows most the code for mounting `Secrets` and `ConfigMaps` to be unified, but may not be the best long-term path.

**Comment by adambkaplan** [comment]
@liggitt feature is proposed for 3.11. API changes are stable, but shouldn't be merged until 3.10 is cut.

**Comment by adambkaplan** [comment]
[(5) Allow using a ConfigMap as an input to a build [builds]](https://trello.com/c/RMKJxJUm/1020-5-allow-using-a-configmap-as-an-input-to-a-build-builds)

**Comment by liggitt** [comment]
/lgtm

/hold
for 3.10 branch cuts

**Pull request merged**